circuits/src/zk_circuits/valid_match_settle_atomic.rs

```diff
@@ -53,13 +54,25 @@ where
         witness: &ValidMatchSettleAtomicWitnessVar<MAX_BALANCES, MAX_ORDERS>,
         cs: &mut PlonkCircuit,
     ) -> Result<(), CircuitError> {
+        // Constrain the malleable fields
+        Self::constrain_malleable_fields(statement, witness, cs)?;
         // Validate the matching engine
         Self::validate_matching_engine(statement, witness, cs)?;
-
         // Validate the internal party's settlement
         Self::validate_settlement(statement, witness, cs)
     }
 
+    /// Constrain the fields that appear in no constraints to prevent them from
+    /// being changed
+    fn constrain_malleable_fields(
+        statement: &ValidMatchSettleAtomicStatementVar<MAX_BALANCES, MAX_ORDERS>,
+        _witness: &ValidMatchSettleAtomicWitnessVar<MAX_BALANCES, MAX_ORDERS>,
+        cs: &mut PlonkCircuit,
+    ) -> Result<(), CircuitError> {
+        // Constrain the relayer fee address
+        NoopGadget::constrain_noop(&statement.relayer_fee_address, cs)
```

Comment:
don't we also have to constrain the external party's relayer fee take to be non-malleable?

Reply:
we do not, because it separately appears in a constraint. Specifically, `statement.external_party_fees.relayer_fee` is constrained to be a valid amount in `validate_external_fees`.